x/mongo/driver/topology/connection.go

```diff
@@ -374,6 +374,17 @@ func (c *connection) readWireMessage(ctx context.Context, dst []byte) ([]byte, e
 		}
 	}
 
+	// In the case of an isMaster response where MaxMessageSize has not yet been set, use a hard-coded
+	// 48000 bytes instead.
+	maxMessageSize := c.desc.MaxMessageSize
+	if maxMessageSize == 0 {
+		maxMessageSize = 48000
+	}
+	if uint32(len(dst)) > maxMessageSize {
```

Comment:
I think this conversion is ok;`len()` returns `int64`, so I could also make sure that `len(dst)` is not greater than the max value of `uint32`...